src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt

```diff
@@ -718,6 +772,13 @@ open class IndexReplicationTask(id: Long, type: String, action: String, descript
                 client.execute(PauseIndexReplicationAction.INSTANCE,
                         PauseIndexReplicationRequest(followerIndexName, TASK_CANCELLATION_REASON))
             }
+
+            // Deleting the follower index if replication is stopped because of leader index deletion
+            if (clusterService.clusterSettings.get(ReplicationPlugin.REPLICATION_REPLICATE_INDEX_DELETION)
+                && currentTaskState.state == ReplicationState.COMPLETED
+                && clusterService.state().metadata.index(leaderIndex) == null)  {
```

Comment:
this check is wrong.

Reply:
Corrected by checking whether leader index is deleted or not.